src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -1129,21 +1130,24 @@ private static unsafe nuint UnalignedCountVector128(ref byte searchSpace)
 
         public static void Reverse(ref byte buf, nuint length)
         {
-            if (Avx2.IsSupported && (nuint)Vector256<byte>.Count * 2 <= length)
+            Debug.Assert(length > 1);
+
+            nint remainder = (nint)length;
+            nint offset = 0;
+
+            // overlapping has a positive performance benefit around 48 elements
+            if (Avx2.IsSupported && remainder >= Vector256<byte>.Count * 1.5)
```

Comment:
This is going to do floating point math. That does not look cheap.

Reply:
I was a bit too fast on the keyboard, didn't check the assembly output. It only evaluates it as a constant if I use `(nint)(Vector256<byte>.Count * 1.5)`

Reply:
> It only evaluates it as a constant if I use (nint)(Vector256<byte>.Count * 1.5)

just `Vector256<byte>.Count * 1.5` also evaluates into a constant, but a FP constant => datasection memory load